Fix issue QueueCommand and AddCommand cannot be found error

1bberto opened the pull request:
With the new `Hangfire.SqlServer 1.8.6` the method `AddCommand` from the class `SqlServerWriteOnlyTransaction`, was changed to `static` https://github.com/HangfireIO/Hangfire/commit/c0d6290153abca71e08834f3133222e0c110063e.

Updating to get the method with the proper `BindingFlags`, adding `BindingFlags.Static`

closes #77

1bberto committed:
fix issue with on SqlTagsTransaction with SqlServerWriteOnlyTransaction

1bberto committed:
Merge branch 'master' of https://github.com/1bberto/Hangfire.Tags into bugfix/hangfire-sql-server

1bberto commented:
@r-win, could you please take a look at this Pull Request, there is an error happening with the new Hangfire.SqlServer version,
thank you

1bberto committed:
Bumped FaceIT.Hangfire.Tags.SqlServer project version to 1.8.3

erwin-faceit commented:
I think this PR is not needed, you need to update to the preview of Hangfire.Tags, which is renamed to FaceIT.Hangfire.Tags.

erwin-faceit commented:
It actually might be that this patch is also required on the hangfire_182 branch, but not on the master branch, because the master branch does NOT support hangfire 1.8+.

In that case, I would prefer it to leave the Hangfire.SqlServer reference alone (to the lowest possible), and use both BindingFlags so that the library remains backwards compatible.

1bberto commented:
Tks @erwin-faceit 
 I will make the change and the target branch

1bberto committed:
Update SqlTagsTransaction.cs

update GetMethod to work with previous versions as well

1bberto commented on src/FaceIT.Hangfire.Tags.SqlServer/SqlTagsTransaction.cs:
```diff
@@ -57,7 +57,7 @@ public SqlTagsTransaction(SqlServerStorageOptions options, IWriteOnlyTransaction
             {
                 _addCommand = transaction.GetType().GetTypeInfo().GetMethod(nameof(AddCommand),
-                    BindingFlags.NonPublic | BindingFlags.Instance);
+                    BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static);
```
Add this to work with `Hangfire.SqlServer` > 1.8.6

1bberto commented:
@erwin-faceit / @r-win  , I've updated the code, moreover I added the other PR to update the branch `hangfire_182` #81, could you please take a look on both?

alvicode commented:
we also need this.
Thanks a lot.

erwin-faceit commented:
This PR will not be merged, since this branch does NOT support Hangfire 1.8+. Thank you for creating the PR in the hangfire_182 branch as well, that works as expected.

@alvicode: I think you're using the FaceIT.Hangfire.Tags packages, which already have this PR included. If you're still using the Hangfire.Tags packages: they are obsolete. I'll try to push stable versions of FaceIT.Hangfire.Tags packages somewhere this week.

vjaiswal-mdsol commented:
Is this issue still persists, Because This PR is not merged

erwin-faceit commented:
No. Do not use Hangfire.Tags with Hangfire >= 1.8.2. Switch to FaceIT.Hangfire.Tags instead.
